src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -0,0 +1,1931 @@
+                System.ReadOnlySpan<string?> M3(string[] arg) => arg;
+                System.ReadOnlySpan<string?> M4(string?[] arg) => arg;
+                System.ReadOnlySpan<int> M5(int?[] arg) => arg;
+                System.ReadOnlySpan<int?> M6(int[] arg) => arg;
+                System.ReadOnlySpan<int?> M7(int?[] arg) => arg;
+                System.ReadOnlySpan<string> M8(string?[] arg) => arg;
+                System.ReadOnlySpan<object> M9(string?[] arg) => arg;
+            }
+            """;
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion)).VerifyDiagnostics(
+            // (5,54): warning CS8619: Nullability of reference types in value of type 'string?[]' doesn't match target type 'string[]'.
+            //     System.ReadOnlySpan<string> M2(string?[] arg) => arg;
+            Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "arg").WithArguments("string?[]", targetType("string")).WithLocation(5, 54),
+            // (8,48): error CS0029: Cannot implicitly convert type 'int?[]' to 'System.ReadOnlySpan<int>'
+            //     System.ReadOnlySpan<int> M5(int?[] arg) => arg;
+            Diagnostic(ErrorCode.ERR_NoImplicitConv, "arg").WithArguments("int?[]", "System.ReadOnlySpan<int>").WithLocation(8, 48),
+            // (9,48): error CS0029: Cannot implicitly convert type 'int[]' to 'System.ReadOnlySpan<int?>'
+            //     System.ReadOnlySpan<int?> M6(int[] arg) => arg;
+            Diagnostic(ErrorCode.ERR_NoImplicitConv, "arg").WithArguments("int[]", "System.ReadOnlySpan<int?>").WithLocation(9, 48),
+            // (11,54): warning CS8619: Nullability of reference types in value of type 'string?[]' doesn't match target type 'string[]'.
+            //     System.ReadOnlySpan<string> M8(string?[] arg) => arg;
+            Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "arg").WithArguments("string?[]", targetType("string")).WithLocation(11, 54),
+            // (12,54): warning CS8619: Nullability of reference types in value of type 'string?[]' doesn't match target type 'object[]'.
+            //     System.ReadOnlySpan<object> M9(string?[] arg) => arg;
+            Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "arg").WithArguments("string?[]", targetType("object")).WithLocation(12, 54));
+
+        string targetType(string inner)
+            => langVersion > LanguageVersion.CSharp12 ? $"System.ReadOnlySpan<{inner}>" : $"{inner}[]";
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Implicit_NullableAnalysis_Nested(LanguageVersion langVersion)
+    {
+        var source = """
+            #nullable enable
+            class C
+            {
+                System.Span<string[]> M1(string[][] arg) => arg;
+                System.Span<string[]> M2(string?[][] arg) => arg;
+                System.Span<string?[]> M3(string[][] arg) => arg;
```

Comment:
>System.Span<string?[]> M3(string[][] arg) => arg; [](http://example.com/codeflow?start=16&length=49)

Are we trying to test the handling of nested nullability or nested arrays? If the former, consider using a generic type rather than an array for element type. For instance:
```csharp
...
Span<S<string?>> M3(S<string>[] arg) => arg;
...

struct S<T> { }
```